packages/scxa-experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -0,0 +1,338 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import URI from 'urijs'
+import { BrowserRouter, Route, NavLink, Switch, Redirect, withRouter } from 'react-router-dom'
+
+import AnatomogramCellTypeHeatmapView from './results/AnatomogramCellTypeHeatmapView'
+
+import TSnePlotView from '@ebi-gene-expression-group/scxa-tsne-plot'
+import { ClustersHeatmapView }  from '@ebi-gene-expression-group/scxa-marker-gene-heatmap'
+
+import BioentityInformation from '@ebi-gene-expression-group/atlas-bioentity-information'
+import { withFetchLoader } from '@ebi-gene-expression-group/atlas-react-fetch-loader'
+
+import {intersection as _intersection, first as _first, map as _map} from 'lodash'
+
+const BioentityInformationWithFetchLoader = withFetchLoader(BioentityInformation)
+
+const RedirectWithSearchAndHash = (props) =>
+  <Redirect to={{ pathname: props.pathname, search: props.location.search, hash: props.location.hash}} />
+
+const RedirectWithLocation = withRouter(RedirectWithSearchAndHash)
+
+const CLUSTERS_PLOT = `clusters`
+const METADATA_PLOT = `metadata`
+const CELL_TYPE_MARKER_GENE_HEATMAP = `celltypes`
+const CLUSTER_MARKER_GENE_HEATMAP = `clusters`
+
+class TSnePlotViewRoute extends React.Component {
+  constructor(props) {
+      super(props)
+    const cellTypeValue = _first(_intersection(_map(this.props.metadata,`label`), this.props.initialCellTypeValues))
+    const search = URI(this.props.location.search).search(true)
+    this.state = {
+      selectedPlotType: Object.keys(this.props.defaultPlotMethodAndParameterisation)[0],
+      geneId: ``,
+      selectedPlotOption: Object.values(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0],
+      selectedPlotOptionLabel: Object.keys(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0] + ": " +
+          Object.values(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0],
+      selectedColourBy: cellTypeValue ? cellTypeValue.toLowerCase() :
+          this.props.ks.length > 0 ? this.props.ks[Math.round((this.props.ks.length -1) / 2)].toString() : ``,
```

Comment:
Same validation checks here as well before passing any function to fetch data
`Object.values(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0]`

We are not sure about default drop down size is grater than 0

Reply:
> Same validation checks here as well before passing any function to fetch data `Object.values(Object.values(this.props.defaultPlotMethodAndParameterisation)[0])[0]`
> 
> We are not sure about default drop down size is grater than 0

Yes, but again, this is not the right place to do validation. We need the value here and I add `defaultPlotMethodAndParameterisation` as a required props. If we don't have this props, we should disable the tab, rather than keeping this value empty as `null`. In another words, the required props we do not validate in the component level anymore.